### PR TITLE
docs: Add mocking useLocation to docs

### PR DESCRIPTION
**Pull request by bigbuckalex**
I added a how to page in the docs that would close(#5190)

I'm not sure if there is a better place for this or if anyone would want to add to it.

### CODE REVIEW

**Comment by jtoar** [comment]
Thanks @bigbuckalex! Yeah this may be too short for a how to. Initially I thought it'd go in the Testing reference, but I know that's quite long. Let me see what others things.

**Comment by KrisCoulson** [comment]
hmm @jtoar I think this is nice a tight. My personal feelings is that How to's don't have to be super long. That being said this could potentially go in to the testing documentation instead of a How To. I think the main thing is this info should index and ideally show up in google when you type "redwoodjs test useParams or useLocation"

File: docs/docs/testing.md [hunk 1]
```diff
@@ -274,6 +274,19 @@ render(<Article article={ title: 'Foobar' } />, {
 })
 ```
 :::
+### Mocking useLocation
+
+To mock ``useLocation`` for your component tests, wrap the component with ``LocationProvider``.
```
Comment (jtoar) [inline docs/docs/testing.md:279]: One set of backticks should do; swap "for" for "in"; colon at the end:

```suggestion
To mock `useLocation` in your component tests, wrap the component with `LocationProvider`:
```

**Review by jtoar — COMMENTED** [review]
Thanks @bigbuckalex, just a few edits and this looks good to me!

**Comment by bigbuckalex** [comment]
Thank you @jtoar! Committed your suggestions.

File: docs/docs/testing.md [hunk 2]
```diff
@@ -274,6 +274,19 @@ render(<Article article={ title: 'Foobar' } />, {
...
+
+To mock `useLocation` in your component tests, wrap the component with `LocationProvider`:
+
+```jsx
+import { LocationProvider } from '@redwoodjs/router'
+
+render(
+<LocationProvider location={{ pathname: '', search: '?cancelled=true' }}>
+   <Component />
+</LocationProvider>
```
Comment (jtoar) [inline docs/docs/testing.md:287]: Last fix: didn't notice the indentation (I'll commit this one)

```suggestion
  <LocationProvider location={{ pathname: '', search: '?cancelled=true' }}>
    <Component />
  </LocationProvider>
```

**Review by jtoar — APPROVED** [review]

**Pull request merged**